apollo-compiler/src/test/graphql/com/example/fragment_in_fragment/AllStarships.kt

```diff
@@ -117,7 +117,7 @@ class AllStarships : Query<AllStarships.Data, AllStarships.Data, Operation.Varia
     }
   }
 
-  data class AllStarship(
+  data class AllStarships1(
```

Comment:
Interesting that it suffixes the class now. I think it should be ok without the `1` since it's nested ? 

```
class AllStarships {
    class AllStarships {
    }
}
```

That should be valid, right ?

Reply:
Seems like it should be, yeah. This should probably be fixed with the 1.3.0 release so you don't risk doing 2 consecutive breaking changes releases. That being said, it's unrelated to this PR so I won't fix it here.

Reply:
Sounds good. We should just remember to fix it in 1.3.0. Maybe it's time we document the coming breaking changes somewhere so we don't forget anything...

Reply:
1.3.0 release checklist there : https://github.com/apollographql/apollo-android/issues/1889

Reply:
It's ok for Kotlin, but not for Java, this is not valid in Java:
```
public class Test {
  static class Test {
    
  }
}
```

We might need to relax our Kotlin version of class name clash resolution and don't include top class into used keywords. But for Java we can't do the same.

Reply:
Thanks for the explanation!